bitcoin/src/address.rs

```diff
@@ -725,15 +756,15 @@ impl<V: NetworkValidation> Address<V> {
         match self.payload {
             Payload::PubkeyHash(_) => Some(AddressType::P2pkh),
             Payload::ScriptHash(_) => Some(AddressType::P2sh),
-            Payload::WitnessProgram { version, program: ref prog } => {
+            Payload::WitnessProgram(ref prog) => {
                 // BIP-141 p2wpkh or p2wsh addresses.
-                match version {
-                    WitnessVersion::V0 => match prog.len() {
+                match prog.version() {
+                    WitnessVersion::V0 => match prog.program().len() {
                         20 => Some(AddressType::P2wpkh),
                         32 => Some(AddressType::P2wsh),
-                        _ => None,
+                        _ => None, // Technically unreachable, but we don't want to panic when not required.
```

Comment:
Silently ignoring bugs is not great either, `debug_assert` would be nice.

Reply:
Agreed. TBH I'd even prefer an unconditional `unreachable!` panic.